src/openrct2/ride/ride.c

```diff
@@ -8226,3 +8226,104 @@ LocationXY16 ride_get_rotated_coords(sint16 x, sint16 y, sint16 z)
 
     return rotatedCoords;
 }
+
+void fix_ride_entrance_locations()
+{
+    sint32 rideIndex;
+    Ride * ride;
+
+    FOR_ALL_RIDES(rideIndex, ride)
+    {
+        for (sint32 stationIndex = 0; stationIndex < MAX_STATIONS; stationIndex++)
+        {
+            LocationXY8 entranceLoc    = ride->entrances[stationIndex];
+            uint8       entranceHeight = ride->station_heights[stationIndex];
+            bool        foundEntrance  = false;
+
+            // Skip if the station has no entrance
+            if (entranceLoc.xy == RCT_XY8_UNDEFINED)
+            {
+                continue;
+            }
+
+            rct_map_element * mapElement = map_get_first_element_at(entranceLoc.x, entranceLoc.y);
+
+            if (mapElement != NULL)
+            {
+                do
+                {
+                    if (map_element_get_type(mapElement) != MAP_ELEMENT_TYPE_ENTRANCE)
+                    {
+                        continue;
+                    }
+                    else if (entranceHeight != mapElement->base_height)
+                    {
+                        continue;
+                    }
+                    else
+                    {
+                        foundEntrance = true;
+                        break;
+                    }
+                }
+                while (!map_element_is_last_for_tile(mapElement++));
+
+                if (foundEntrance)
+                {
+                    continue;
+                }
+            }
```

Comment:
I recommend extracting this to a new function in map called `map_get_ride_entrance_element_at`. Similar to the already existing `map_get_park_entrance_element_at` function. You can then remove the need for the `foundEntrance` flag.

Reply:
Done this. I based the function on map_get_park_entrance_element_at, which also eliminated a potential bug where it would detect an exit or park entrance as a ride entrance.